odl/solvers/util/steplen.py

```diff
@@ -195,21 +196,48 @@ def __init__(self, constant):
     def __call__(self, x, direction, dir_derivative):
         """Calculate the step length at a point.
 
+        All arguments are ignored and are only added to fit the interface.
+        """
+        return self.constant
+
+
+class PredefinedLineSearch(LineSearch):
+
+    """Line search object that returns a step length from a function."""
+
+    def __init__(self, func):
+        """Initialize a new instance.
+
         Parameters
         ----------
-        x : `LinearSpaceElement`
-            The current point
-        direction : `LinearSpaceElement`
-            Search direction in which the line search should be computed
-        dir_derivative : float
-            Directional derivative along the ``direction``
+        func : callable
+            Function that when called with an iteration count. Iterations are
+            zero indexed, i.e. the first iteration is the zero:th.
+
+        Examples
+        --------
+        Make a step size that is 1.0 for the first 5 iterations, then 0.1
+
+        >>> def step_length(iter):
+        ...     if iter < 5:
+        ...         return 1.0
+        ...     else:
+        ...         return 0.1
+        >>> line_search = PredefinedLineSearch(step_length)
+        """
+        if not callable(func):
+            raise TypeError('`func` must be a callable.')
```

Comment:
The more I see these checks, the more I doubt their value, in particular in cases like this where the failure will come immediately anyway. The only gain here is that the user knows from the message which argument was not callable and that it's not an internal bug that causes the failure.

It's just a thought that popped into my mind, no need to change anything here. But we should keep in mind not to overdo it, otherwise we have redundant checks all over the place.